examples/creatives/banner_nonlinear.js

```diff
@@ -3,36 +3,114 @@ const X_OFFSET_PERCENTAGE = .15;
 const WIDTH_PERCENTAGE = .7;
 const HEIGHT_PERCENTAGE = .15;
 
+/**
+ * A sample SIMID non-linear ad for a banner ad that shows a website when clicked on.
+ * P.S: Not all websites can be shown and they would need to allow the x-frame-options
+ *      header to sameorigin for the ad to display correctly. More information here:
+ *      https://web.dev/samesite-cookies-explained/
+ */
 class BannerNonLinear extends BaseSimidCreative {
     constructor() {
         super();
+
+        /**
+         * The desired text on the banner.
+         * @private {?string}
+         */
+        this.bannerText_ = null;
+        
+        /**
+         * The web URL to be displayed.
+         * @private {?string}
+         */
+        this.webUrl_ = null;
+
         this.addButtonClickActions_();
     }
 
     /** @override */
     onInit(eventData) {
-        super.onInit(eventData);
+        this.updateInternalOnInit(eventData);
+        this.validateAndParseAdParams_(eventData);
+        this.updateCreativeWithParams_();
         this.dynamicResize_();
```

Comment:
If init gets rejected, you probably don't want to continue on to call updateCreativeWithParams and dynamicResize

Reply:
looking at the other creatives and how they were wrote, they all finish executing that function even if it gets rejected. Is there a way to do what you said without adding too much complexity to the creatives?

Reply:
hmm, if the others do it then I guess its fine to do this just to keep it consistent.  I guess if init is rejected the iframe will just never show so it doesn't really matter what else you do